internal/core/data/v2/application/event.go

```diff
@@ -93,7 +95,8 @@ func PublishEvent(addEventReq dto.AddEventRequest, profileName string, deviceNam
 		addEventReq.Event.Readings[index].Versionable = common.NewVersionable()
 	}
 
-	data, err = json.Marshal(addEventReq)
+	//marshal AddEventRequest DTO to byte array per contentType
+	data, err = marshal(addEventReq, clients.FromContext(ctx, clients.ContentType))
```

Comment:
Problem here  with CBOR (and actually with JSON also) is we are re-encoding the request DTO. If no error occurred, we know the request received is good so is there a way we can just pass PublishEvent() the encoded data?

I recommend that `ReadAddEventRequest` also return the encoded []byte data for the request that can then be passed to PublishEvent(). 

@jpwhitemn,  Then the question for Core WG is do we want the Publish to be async like it is in V1 for CBOR, i.e. don't wait for validation and persistence. Just get it onto the MessageBus ASAP.

Reply:
@lenny-intel, re-encoding the request DTO is a necessary step here for both CBOR and JSON as we set ApiVerion on the request DTO prior to publishing to MessageBus.  Your recommendation will work if we move the code of setting ApiVersion to `ReadAddEventRequest`.

Reply:
@judehung , `ApiVersion` is now required in the request DTO , so it will not pass validation if not already present. Also, it is already set in the factory method. We should be able to Publish the original encoded data if no errors occurred.

Since the app services can handle the request directly from Device Services, the request is also validated, so we could publish async to Core data processing. @jpwhitemn we should discuss this in Core WG.

Reply:
Per Core WG - DECISION: publish initially encoded object to (not re-encoding) async to message bus (app service) while simultaneously validate/persist in (separate go function)
If persistence fails, it does not impact message publication (cow is already out of the barn)
Let app service (or whatever is on the other end of the message bus) deal with bad/invalid data